pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java

```diff
@@ -101,6 +111,9 @@ public void initChannel(SocketChannel ch) throws Exception {
                 ch.pipeline().addLast("handler", new ClientCnx(client));
             }
         });
+
+        this.dnsResolver = new DnsNameResolverBuilder(eventLoopGroup.next())
+                .channelType(EventLoopUtil.getDatagramChannelClass(eventLoopGroup)).build();
```

Comment:
should we add `queryTimeoutMillis(2000)`?

Reply:
and also `traceEnabled(true)` that can help for any resolution failure?

Reply:
There's already a default timeout of 5secs. We can tune it down if desired